### ISSUE TITLE
[2.1.x]error: Assertion failed: [ptr_] 

### ISSUE THREAD

**Issue by neumino** [rethinkdb/rethinkdb#4685]
Just ran `thinky` test 3 times in a row, and got this crash.

```
michel@h9:~/projects/rethinkdb-all/rethinkdb/build/debug$ rm -R rethinkdb_data; ./rethinkdb
Recursively removing directory /home/michel/projects/rethinkdb-all/rethinkdb/build/debug/rethinkdb_data/tmp
Initializing directory /home/michel/projects/rethinkdb-all/rethinkdb/build/debug/rethinkdb_data
Running rethinkdb 2.1.0-0BETA2-153-g026591 (debug) (GCC 5.2.0)...
Running on Linux 4.0.7-2-ARCH x86_64
Loading data from directory /home/michel/projects/rethinkdb-all/rethinkdb/build/debug/rethinkdb_data
Our server ID is 034e4853-6340-48a0-9d1d-d4c4e5775427
Listening for intracluster connections on port 29015
Listening for client driver connections on port 28015
Listening for administrative HTTP connections on port 8080
Listening on addresses: 127.0.0.1, ::1
To fully expose RethinkDB on the network, bind to all addresses by running rethinkdb with the `--bind all` command line option.
Server ready, "h9_9im" 034e4853-6340-48a0-9d1d-d4c4e5775427
debug: Finished checking for newer versions of RethinkDB. We are already running the most up-to-date version.
Version: rethinkdb 2.1.0-0BETA2-153-g026591 (debug) (GCC 5.2.0)
error: Error in ./src/containers/scoped.hpp at line 93:
error: Assertion failed: [ptr_] 
error: Backtrace:
error: Wed Aug 12 21:25:55 2015
       
       1: rethinkdb_backtrace(void**, int) at rethinkdb_backtrace.cc:101
       2: backtrace_t::backtrace_t() at backtrace.cc:203 (discriminator 1)
       3: lazy_backtrace_formatter_t::lazy_backtrace_formatter_t() at backtrace.cc:283
       4: format_backtrace(bool) at backtrace.cc:198
       5: report_fatal_error(char const*, int, char const*, ...) at errors.cc:83 (discriminator 1)
       6: scoped_ptr_t<serializer_t>::get() const at scoped.hpp:93 (discriminator 1)
       7: real_multistore_ptr_t::get_serializer() at table_interface.cc:148
       8: real_table_persistence_interface_t::is_gc_active() const at table_interface.cc:343
       9: jobs_manager_t::on_get_job_reports(signal_t*, mailbox_addr_t<void (std::vector<query_job_report_t, std::allocator<query_job_report_t> >, std::vector<disk_compaction_job_report_t, std::allocator<disk_compaction_job_report_t> >, std::vector<index_construction_job_report_t, std::allocator<index_construction_job_report_t> >, std::vector<backfill_job_report_t, std::allocator<backfill_job_report_t> >)> const&) at manager.cc:108
       10: void std::_Mem_fn_base<void (jobs_manager_t::*)(signal_t*, mailbox_addr_t<void (std::vector<query_job_report_t, std::allocator<query_job_report_t> >, std::vector<disk_compaction_job_report_t, std::allocator<disk_compaction_job_report_t> >, std::vector<index_construction_job_report_t, std::allocator<index_construction_job_report_t> >, std::vector<backfill_job_report_t, std::allocator<backfill_job_report_t> >)> const&), true>::operator()<signal_t*, mailbox_addr_t<void (std::vector<query_job_report_t, std::allocator<query_job_report_t> >, std::vector<disk_compaction_job_report_t, std::allocator<disk_compaction_job_report_t> >, std::vector<index_construction_job_report_t, std::allocator<index_construction_job_report_t> >, std::vector<backfill_job_report_t, std::allocator<backfill_job_report_t> >)>, void>(jobs_manager_t*, signal_t*&&, mailbox_addr_t<void (std::vector<query_job_report_t, std::allocator<query_job_report_t> >, std::vector<disk_compaction_job_report_t, std::allocator<disk_compaction_job_report_t> >, std::vector<index_construction_job_report_t, std::allocator<index_construction_job_report_t> >, std::vector<backfill_job_report_t, std::allocator<backfill_job_report_t> >)>&&) const at functional:600 (discriminator 4)
       11: void std::_Bind<std::_Mem_fn<void (jobs_manager_t::*)(signal_t*, mailbox_addr_t<void (std::vector<query_job_report_t, std::allocator<query_job_report_t> >, std::vector<disk_compaction_job_report_t, std::allocator<disk_compaction_job_report_t> >, std::vector<index_construction_job_report_t, std::allocator<index_construction_job_report_t> >, std::vector<backfill_job_report_t, std::allocator<backfill_job_report_t> >)> const&)> (jobs_manager_t*, std::_Placeholder<1>, std::_Placeholder<2>)>::__call<void, signal_t*&&, mailbox_addr_t<void (std::vector<query_job_report_t, std::allocator<query_job_report_t> >, std::vector<disk_compaction_job_report_t, std::allocator<disk_compaction_job_report_t> >, std::vector<index_construction_job_report_t, std::allocator<index_construction_job_report_t> >, std::vector<backfill_job_report_t, std::allocator<backfill_job_report_t> >)>&&, 0ul, 1ul, 2ul>(std::tuple<signal_t*&&, mailbox_addr_t<void (std::vector<query_job_report_t, std::allocator<query_job_report_t> >, std::vector<disk_compaction_job_report_t, std::allocator<disk_compaction_job_report_t> >, std::vector<index_construction_job_report_t, std::allocator<index_construction_job_report_t> >, std::vector<backfill_job_report_t, std::allocator<backfill_job_report_t> >)>&&>&&, std::_Index_tuple<0ul, 1ul, 2ul>) at functional:1075
       12: void std::_Bind<std::_Mem_fn<void (jobs_manager_t::*)(signal_t*, mailbox_addr_t<void (std::vector<query_job_report_t, std::allocator<query_job_report_t> >, std::vector<disk_compaction_job_report_t, std::allocator<disk_compaction_job_report_t> >, std::vector<index_construction_job_report_t, std::allocator<index_construction_job_report_t> >, std::vector<backfill_job_report_t, std::allocator<backfill_job_report_t> >)> const&)> (jobs_manager_t*, std::_Placeholder<1>, std::_Placeholder<2>)>::operator()<signal_t*, mailbox_addr_t<void (std::vector<query_job_report_t, std::allocator<query_job_report_t> >, std::vector<disk_compaction_job_report_t, std::allocator<disk_compaction_job_report_t> >, std::vector<index_construction_job_report_t, std::allocator<index_construction_job_report_t> >, std::vector<backfill_job_report_t, std::allocator<backfill_job_report_t> >)>, void>(signal_t*&&, mailbox_addr_t<void (std::vector<query_job_report_t, std::allocator<query_job_report_t> >, std::vector<disk_compaction_job_report_t, std::allocator<disk_compaction_job_report_t> >, std::vector<index_construction_job_report_t, std::allocator<index_construction_job_report_t> >, std::vector<backfill_job_report_t, std::allocator<backfill_job_report_t> >)>&&) at functional:1133
       13: std::_Function_handler<void (signal_t*, mailbox_addr_t<void (std::vector<query_job_report_t, std::allocator<query_job_report_t> >, std::vector<disk_compaction_job_report_t, std::allocator<disk_compaction_job_report_t> >, std::vector<index_construction_job_report_t, std::allocator<index_construction_job_report_t> >, std::vector<backfill_job_report_t, std::allocator<backfill_job_report_t> >)>), std::_Bind<std::_Mem_fn<void (jobs_manager_t::*)(signal_t*, mailbox_addr_t<void (std::vector<query_job_report_t, std::allocator<query_job_report_t> >, std::vector<disk_compaction_job_report_t, std::allocator<disk_compaction_job_report_t> >, std::vector<index_construction_job_report_t, std::allocator<index_construction_job_report_t> >, std::vector<backfill_job_report_t, std::allocator<backfill_job_report_t> >)> const&)> (jobs_manager_t*, std::_Placeholder<1>, std::_Placeholder<2>)> >::_M_invoke(std::_Any_data const&, signal_t*&&, mailbox_addr_t<void (std::vector<query_job_report_t, std::allocator<query_job_report_t> >, std::vector<disk_compaction_job_report_t, std::allocator<disk_compaction_job_report_t> >, std::vector<index_construction_job_report_t, std::allocator<index_construction_job_report_t> >, std::vector<backfill_job_report_t, std::allocator<backfill_job_report_t> >)>&&) at functional:1873
       14: std::function<void (signal_t*, mailbox_addr_t<void (std::vector<query_job_report_t, std::allocator<query_job_report_t> >, std::vector<disk_compaction_job_report_t, std::allocator<disk_compaction_job_report_t> >, std::vector<index_construction_job_report_t, std::allocator<index_construction_job_report_t> >, std::vector<backfill_job_report_t, std::allocator<backfill_job_report_t> >)>)>::operator()(signal_t*, mailbox_addr_t<void (std::vector<query_job_report_t, std::allocator<query_job_report_t> >, std::vector<disk_compaction_job_report_t, std::allocator<disk_compaction_job_report_t> >, std::vector<index_construction_job_report_t, std::allocator<index_construction_job_report_t> >, std::vector<backfill_job_report_t, std::allocator<backfill_job_report_t> >)>) const at functional:2272
       15: mailbox_t<void (mailbox_addr_t<void (std::vector<query_job_report_t, std::allocator<query_job_report_t> >, std::vector<disk_compaction_job_report_t, std::allocator<disk_compaction_job_report_t> >, std::vector<index_construction_job_report_t, std::allocator<index_construction_job_report_t> >, std::vector<backfill_job_report_t, std::allocator<backfill_job_report_t> >)>)>::read_impl_t::read(read_stream_t*, signal_t*) at typed.hpp:169
       16: mailbox_manager_t::mailbox_read_coroutine(connectivity_cluster_t::connection_t*, auto_drainer_t::lock_t, threadnum_t, unsigned long, std::vector<char, std::allocator<char> >*, long, mailbox_manager_t::force_yield_t) at mailbox.cc:276
       17: ./rethinkdb() [0x1ff5876] at 0x1ff5876 ()
       18: ./rethinkdb() [0x1ff67a0] at 0x1ff67a0 ()
       19: callable_action_wrapper_t::run() at runtime_utils.cc:43
       20: coro_t::run() at coroutines.cc:207
       21: ./rethinkdb() [0x1ff61ad] at 0x1ff61ad ()
       22: mailbox_manager_t::on_local_message(connectivity_cluster_t::connection_t*, auto_drainer_t::lock_t, std::vector<char, std::allocator<char> >&&) at mailbox.cc:212
       23: connectivity_cluster_t::send_message(connectivity_cluster_t::connection_t*, auto_drainer_t::lock_t, unsigned char, cluster_send_message_write_callback_t*) at cluster.cc:1338
       24: send(mailbox_manager_t*, raw_mailbox_t::address_t, mailbox_write_callback_t*) at mailbox.cc:128
       25: void send<mailbox_addr_t<void (std::vector<query_job_report_t, std::allocator<query_job_report_t> >, std::vector<disk_compaction_job_report_t, std::allocator<disk_compaction_job_report_t> >, std::vector<index_construction_job_report_t, std::allocator<index_construction_job_report_t> >, std::vector<backfill_job_report_t, std::allocator<backfill_job_report_t> >)> >(mailbox_manager_t*, mailbox_t<void (mailbox_addr_t<void (std::vector<query_job_report_t, std::allocator<query_job_report_t> >, std::vector<disk_compaction_job_report_t, std::allocator<disk_compaction_job_report_t> >, std::vector<index_construction_job_report_t, std::allocator<index_construction_job_report_t> >, std::vector<backfill_job_report_t, std::allocator<backfill_job_report_t> >)>)>::address_t, mailbox_addr_t<void (std::vector<query_job_report_t, std::allocator<query_job_report_t> >, std::vector<disk_compaction_job_report_t, std::allocator<disk_compaction_job_report_t> >, std::vector<index_construction_job_report_t, std::allocator<index_construction_job_report_t> >, std::vector<backfill_job_report_t, std::allocator<backfill_job_report_t> >)> const&) at typed.hpp:206 (discriminator 3)
       26: ./rethinkdb() [0x1e33de3] at 0x1e33de3 ()
       27: ./rethinkdb() [0x1e35baa] at 0x1e35baa ()
       28: ./rethinkdb() [0x1e35b1c] at 0x1e35b1c ()
       29: callable_action_wrapper_t::run() at runtime_utils.cc:43
       30: coro_t::run() at coroutines.cc:207
       31: ./rethinkdb() [0x1e354af] at 0x1e354af ()
       32: ./rethinkdb() [0x1e35187] at 0x1e35187 ()
       33: ./rethinkdb() [0x1e34d67] at 0x1e34d67 ()
       34: jobs_artificial_table_backend_t::get_all_job_reports(signal_t*, std::map<uuid_u, ql::datum_t, std::less<uuid_u>, std::allocator<std::pair<uuid_u const, ql::datum_t> > >*) at backend.cc:103
       35: jobs_artificial_table_backend_t::read_all_rows_as_vector(signal_t*, std::vector<ql::datum_t, std::allocator<ql::datum_t> >*, admin_err_t*) at backend.cc:132
       36: artificial_table_backend_t::read_all_rows_as_stream(ql::backtrace_id_t, ql::datum_range_t const&, sorting_t, signal_t*, counted_t<ql::datum_stream_t>*, admin_err_t*) at backend.cc:15
       37: artificial_table_t::read_all(ql::env_t*, std::string const&, ql::backtrace_id_t, std::string const&, ql::datum_range_t const&, sorting_t, read_mode_t) at artificial_table.cc:74
       38: ql::table_t::as_seq(ql::env_t*, std::string const&, ql::backtrace_id_t, ql::datum_range_t const&, sorting_t) at val.cc:217
       39: ql::table_slice_t::as_seq(ql::env_t*, ql::backtrace_id_t) at val.cc:174 (discriminator 4)
       40: ql::val_t::as_selection(ql::env_t*) at val.cc:606 (discriminator 1)
       41: ql::filter_term_t::eval_impl(ql::scope_env_t*, ql::args_t*, ql::eval_flags_t) const at seq.cc:285
       42: ql::op_term_t::term_eval(ql::scope_env_t*, ql::eval_flags_t) const at op.cc:225
       43: ql::runtime_term_t::eval(ql::scope_env_t*, ql::eval_flags_t) const at term.cc:353
       44: ql::op_term_t::maybe_grouped_data(ql::scope_env_t*, ql::argvec_t*, ql::eval_flags_t, counted_t<ql::grouped_data_t>*, scoped_ptr_t<ql::val_t>*) const at op.cc:303 (discriminator 2)
       45: ql::op_term_t::term_eval(ql::scope_env_t*, ql::eval_flags_t) const at op.cc:211
       46: ql::runtime_term_t::eval(ql::scope_env_t*, ql::eval_flags_t) const at term.cc:353
       47: ql::op_term_t::maybe_grouped_data(ql::scope_env_t*, ql::argvec_t*, ql::eval_flags_t, counted_t<ql::grouped_data_t>*, scoped_ptr_t<ql::val_t>*) const at op.cc:303 (discriminator 2)
       48: ql::op_term_t::term_eval(ql::scope_env_t*, ql::eval_flags_t) const at op.cc:211
       49: ql::runtime_term_t::eval(ql::scope_env_t*, ql::eval_flags_t) const at term.cc:353
       50: ql::op_term_t::maybe_grouped_data(ql::scope_env_t*, ql::argvec_t*, ql::eval_flags_t, counted_t<ql::grouped_data_t>*, scoped_ptr_t<ql::val_t>*) const at op.cc:303 (discriminator 2)
       51: ql::op_term_t::term_eval(ql::scope_env_t*, ql::eval_flags_t) const at op.cc:211
       52: ql::runtime_term_t::eval(ql::scope_env_t*, ql::eval_flags_t) const at term.cc:353
       53: ql::make_obj_term_t::term_eval(ql::scope_env_t*, ql::eval_flags_t) const at datum_terms.cc:87 (discriminator 1)
       54: ql::runtime_term_t::eval(ql::scope_env_t*, ql::eval_flags_t) const at term.cc:353
       55: ql::make_obj_term_t::term_eval(ql::scope_env_t*, ql::eval_flags_t) const at datum_terms.cc:87 (discriminator 1)
       56: ql::runtime_term_t::eval(ql::scope_env_t*, ql::eval_flags_t) const at term.cc:353
       57: ql::query_cache_t::ref_t::run(ql::env_t*, Response*) at query_cache.cc:302
       58: ql::query_cache_t::ref_t::fill_response(Response*) at query_cache.cc:248
       59: ql::run(ql::query_id_t&&, ql::protob_t<Query>, Response*, ql::query_cache_t*, signal_t*) at term.cc:268
       60: rdb_query_server_t::run_query(ql::query_id_t&&, ql::protob_t<Query> const&, Response*, ql::query_cache_t*, signal_t*) at query_server.cc:49 (discriminator 3)
       61: query_server_t::handle(http_req_t const&, http_res_t*, signal_t*) at protob.cc:755
       62: routing_http_app_t::handle(http_req_t const&, http_res_t*, signal_t*) at routing_app.cc:29 (discriminator 4)
       63: routing_http_app_t::handle(http_req_t const&, http_res_t*, signal_t*) at routing_app.cc:29 (discriminator 4)
       64: http_server_t::handle_conn(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t) at http.cc:413
       65: boost::_mfi::mf2<void, http_server_t, scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t>::operator()(http_server_t*, scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t) const at mem_fn_template.hpp:280 (discriminator 6)
       66: void boost::_bi::list3<boost::_bi::value<http_server_t*>, boost::arg<1>, boost::_bi::value<auto_drainer_t::lock_t> >::operator()<boost::_mfi::mf2<void, http_server_t, scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t>, boost::_bi::list1<scoped_ptr_t<linux_tcp_conn_descriptor_t>&> >(boost::_bi::type<void>, boost::_mfi::mf2<void, http_server_t, scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t>&, boost::_bi::list1<scoped_ptr_t<linux_tcp_conn_descriptor_t>&>&, int) at bind.hpp:394 (discriminator 4)
       67: void boost::_bi::bind_t<void, boost::_mfi::mf2<void, http_server_t, scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t>, boost::_bi::list3<boost::_bi::value<http_server_t*>, boost::arg<1>, boost::_bi::value<auto_drainer_t::lock_t> > >::operator()<scoped_ptr_t<linux_tcp_conn_descriptor_t>&>(scoped_ptr_t<linux_tcp_conn_descriptor_t>&) at bind.hpp:907
       68: std::_Function_handler<void (scoped_ptr_t<linux_tcp_conn_descriptor_t>&), boost::_bi::bind_t<void, boost::_mfi::mf2<void, http_server_t, scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t>, boost::_bi::list3<boost::_bi::value<http_server_t*>, boost::arg<1>, boost::_bi::value<auto_drainer_t::lock_t> > > >::_M_invoke(std::_Any_data const&, scoped_ptr_t<linux_tcp_conn_descriptor_t>&) at functional:1873
       69: std::function<void (scoped_ptr_t<linux_tcp_conn_descriptor_t>&)>::operator()(scoped_ptr_t<linux_tcp_conn_descriptor_t>&) const at functional:2272
       70: linux_nonthrowing_tcp_listener_t::handle(int) at network.cc:923
       71: void std::_Mem_fn_base<void (linux_nonthrowing_tcp_listener_t::*)(int), true>::operator()<int&, void>(linux_nonthrowing_tcp_listener_t*, int&) const at functional:600 (discriminator 4)
       72: void std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)>::__call<void, , 0ul, 1ul>(std::tuple<>&&, std::_Index_tuple<0ul, 1ul>) at functional:1075
       73: void std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)>::operator()<, void>() at functional:1133
       74: callable_action_instance_t<std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)> >::run_action() at callable_action.hpp:31
       75: callable_action_wrapper_t::run() at runtime_utils.cc:43
       76: coro_t::run() at coroutines.cc:207
       77: void coro_t::spawn_now_dangerously<std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)> >(std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)>&&) at coroutines.hpp:52
       78: linux_nonthrowing_tcp_listener_t::accept_loop(auto_drainer_t::lock_t) at network.cc:892 (discriminator 4)
       79: void std::_Mem_fn_base<void (linux_nonthrowing_tcp_listener_t::*)(auto_drainer_t::lock_t), true>::operator()<auto_drainer_t::lock_t&, void>(linux_nonthrowing_tcp_listener_t*, auto_drainer_t::lock_t&) const at functional:600 (discriminator 6)
       80: void std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(auto_drainer_t::lock_t)> (linux_nonthrowing_tcp_listener_t*, auto_drainer_t::lock_t)>::__call<void, , 0ul, 1ul>(std::tuple<>&&, std::_Index_tuple<0ul, 1ul>) at functional:1075
       81: void std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(auto_drainer_t::lock_t)> (linux_nonthrowing_tcp_listener_t*, auto_drainer_t::lock_t)>::operator()<, void>() at functional:1133
       82: callable_action_instance_t<std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(auto_drainer_t::lock_t)> (linux_nonthrowing_tcp_listener_t*, auto_drainer_t::lock_t)> >::run_action() at callable_action.hpp:31
       83: callable_action_wrapper_t::run() at runtime_utils.cc:43
       84: coro_t::run() at coroutines.cc:207
error: Exiting.
Trace/breakpoint trap (core dumped)
```

```
michel@h9:~/projects/rethinkdb-all/rethinkdb$ ./configure --allow-fetch
* Detecting system configuration
Bash:                           4.3.39(1)-release
Use ccache:                     no
C++ Compiler:                   GCC 5.2.0 (/usr/bin/c++)
Host System:                    x86_64-unknown-linux-gnu
Build System:                   Linux 4.0.7-2-ARCH x86_64
Cross-compiling:                no
Host Operating System:          Linux
Build Architecture:             x86_64
C++11:                          ok
Precompiled web assets:         no
Protobuf compiler:              /usr/bin/protoc
python:                         python 2.7.10
Node.js package manager:        /usr/bin/npm
coffee:                         external/coffee-script_1.7.1
Browserify:                     external/browserify_3.24.13
bluebird:                       external/bluebird_2.9.32
web UI dependencies:            external/admin-deps_2.0.3
wget:                           /usr/bin/wget
curl:                           /usr/bin/curl
Google Test:                    external/gtest_1.6.0
termcap:                        -ltermcap
boost_system:                   -lboost_system
ssl:                            -lssl
protobuf:                       -lprotobuf
icui18n:                        -licui18n
icuuc:                          -licuuc
icudata:                        -licudata
v8 javascript engine:           external/v8_3.30.33.16
RE2:                            external/re2_20140111
z:                              -lz
crypto:                         -lcrypto
curl:                           -lcurl
malloc:                         jemalloc
jemalloc (static):              external/jemalloc_3.6.0
Test protobuf:                  ok
Test boost:                     ok
Installation prefix:            /usr/local
Configuration prefix:           /usr/local/etc
Runtime data prefix:            /usr/local/var
* Wrote configuration to config.mk
```

I'm on branch 2.1.x and commit 026591a02e0c35ab12696704b110bc3ebe837ae5

**Comment by neumino** [comment]
Not sure if it's relevant, but the web ui tells me that I have about 200 tables and 250 indexes.

**Comment by danielmewes** [comment]
Thanks @neumino !
Looking at the code this looks like a debug-only problem. We check whether the pointer is null, but only after calling `get()` on a `scoped_ptr_t`. In debug mode the call to `get()` triggers the assertion. In release mode it will work as expected and return null, which is then handled further up the call chain.

**Comment by danielmewes** [comment]
That being said I think there's another bug close to this that could cause a seg fault (maybe even #4675?). Looking into that now.

**Comment by VeXocide** [comment]
I'm afraid this is my bad, the fix should be as simple as
```
diff --git a/src/clustering/administration/persist/table_interface.cc b/src/clustering/administration/persist/table_interface.cc
index 8e98493..9da74c4 100644
--- a/src/clustering/administration/persist/table_interface.cc
+++ b/src/clustering/administration/persist/table_interface.cc
@@ -144,7 +144,7 @@ public:
     }
 
     serializer_t *get_serializer() {
-        return serializer.get();
+        return serializer.get_or_null();
     }
 
     store_view_t *get_cpu_sharded_store(size_t i) {
```

It shouldn't explain the segfault though I explicitly check for the `nullptr` in `is_gc_active`
```
            serializer_t *serializer =
                real_multistore.second.first->get_serializer();
            if (serializer == nullptr || ...
```

**Comment by danielmewes** [comment]
Yes a segfault could happen as far as I can tell because in `~real_multistore_ptr_t` we destruct the serializer first, before we drain `drainer`.

`real_table_persistence_interface_t::is_gc_active()` appears to assume that the drainer lock is enough to make sure that the serializer pointer it got from `get_serializer()` remains valid.

Not directly related, just noticed it when looking at the code.

I might have been overly optimistic though to suggest that this could be an explanation for #4675. Do we call `is_gc_active()` only for the jobs table?

**Comment by VeXocide** [comment]
Yep, `is_gc_active()` is only called through `rethinkdb.jobs`.

**Comment by VeXocide** [comment]
In CR 3153 by @danielmewes.

**Comment by VeXocide** [comment]
Merged into branches `next` and `v2.1.x` via commits d011e80eecb36203a1ebeb63d94b448e8576a82e and 57847c1da46f58e7cd4949210c9ebd7ae6d6c2a0 respectively.
